### DISCUSSION TITLE
Stuck at "Attaching to dyno..." (now solved)

### DISCUSSION THREAD

**Discussion by RebbePod** [aahnik/tgcf#74] [Q&A] [answered]
Followed the steps to deploy to heroku, after running tg-login in the console, it seems to be "Attaching to dyno..." forever.
It didn't ask for my phone number or bot token.

**Comment by aahnik** [comment]
Yes, I am aware of this. @RebbePod 

heroku's system is a bit weird. and problematic. I will provide an alternate way soon.

**Reply by RebbePod** [reply]
No problem, thanks!

**Reply by aahnik** [reply]
hi @RebbePod , i [updated the guide for Heroku](https://github.com/aahnik/tgcf/wiki/Deploy-to-Heroku).

read this https://github.com/aahnik/tgcf/wiki/Login-with-a-bot-or-user-account#generate-session-string

note:

running the console in heroku is unnecessary extra hassle. so no need of it. install tg-login in your computer/android, and generate session string.

**Reply by aahnik** [reply]
have you run `pip install tg-login` ?

**Reply by aahnik** [reply]
you may read this : https://github.com/aahnik/tgcf/wiki/Run-tgcf-on-Windows

**Comment by aahnik** [comment, marked as answer]
I made a very secure repl. you can run it online and generate a session string.

read this https://github.com/aahnik/tgcf/wiki/Login-with-a-bot-or-user-account#run-online

but running in your own computer/android is better.
